api/client.go

```diff
@@ -86,7 +86,7 @@ func get[Res any](client Client, path string) (*Res, error) {
 	return &res, nil
 }
 
-func list[Res any](client Client, path string, query map[string][]string) ([]Res, error) {
+func list[Res any](client Client, path string, query map[string][]string) (*Res, error) {
```

Comment:
This change makes `list` very similar to `get` with minor differences around query parameters. It can probably be merged into a single func at some point